src/impls.rs

```diff
@@ -252,3 +262,50 @@ where
         }
     }
 }
+
+#[cfg(feature = "alloc")]
+mod if_alloc {
+    use crate::{Format, Formatter};
+
+    impl<T> Format for alloc::boxed::Box<T>
+    where
+        T: ?Sized + Format,
+    {
+        fn format(&self, f: &mut Formatter) {
+            self.as_ref().format(f)
```

Comment:
I would suggest using UFCS syntax (or whatever it was called) here to make sure we calling `T`'s format method and not recursively calling `Box<T>`'s. That is:

``` rust
T::format(self, f)
```

The first argument may need to be `&**self` or something like that. This comment applies to the other `Format` implementations.